custom/enikshay/tasks.py

```diff
@@ -418,18 +396,28 @@ def get_aggregated_scores(self, latest_adherence_date, adherence_schedule_date_s
 
         return update
 
-    def check_if_needs_update(self, case_properties_expected):
+    def check_and_return(self, update_dict):
         """
         Args:
-            case_properties_expected: dict of case property name to values
+            update_dict: dict of case property name to values
 
         Returns:
-            True if any one of case_properties_expected is not set on self.episode case
+            Checks if any one of update_dict is not set on self.episode case,
+                if any of them are not returns update_dict after formatting any
+                date values to string.
+                If all of them are set as expected, returns None
         """
-        return any([
+        needs_update = any([
             self.get_property(k) != v
-            for (k, v) in case_properties_expected.iteritems()
+            for (k, v) in update_dict.iteritems()
         ])
+        if needs_update:
+            # cast any date objects to strings, so that Django template rendering
+            # used by update_case formats dates as YYYY:MM:DD instead of (Month Date, Year)
+            return {key: str(val) if isinstance(val, datetime.date) else val
```

Comment:
i just made a [PR](https://github.com/dimagi/commcare-hq/pull/16577) which makes the update_case utils use CaseBlock so that can just reuse the formatting done within the CaseBlock util (something i've wanted to do for some time).

once that's merged i think we can just merge master into here and remove any special formatting done so that the formatting is done in one place only